crates/asset-importer-rs-post-process/src/steps/flip_uvs.rs

```diff
@@ -1,20 +1,209 @@
 use asset_importer_rs_core::{AiPostProcess, AiPostProcessSteps};
-use asset_importer_rs_scene::AiScene;
+use asset_importer_rs_scene::{
+    AiMaterial, AiMesh, AiPropertyTypeInfo, AiScene, AiUvTransform, matkey,
+};
+use bytemuck;
 use enumflags2::BitFlags;
 
+#[derive(Debug, PartialEq)]
+pub enum FlipUVsError {
+    UvsNotClippedError(String, usize),
+}
+impl std::fmt::Display for FlipUVsError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            FlipUVsError::UvsNotClippedError(channel_name, channel_index) => {
+                write!(
+                    f,
+                    "UVs in channel named ({}) and with channel index ({}) do not have all their uv components between 0.0 and 1.0 ",
+                    channel_name, channel_index
+                )
+            }
+        }
+    }
+}
+impl std::error::Error for FlipUVsError {}
+
+#[derive(Copy, Clone)]
+pub enum UvFlipVariant {
+    X,
+    Y,
+    Z,
+}
+impl Default for UvFlipVariant {
+    fn default() -> Self {
+        Self::Y
+    }
+}
+
 /// Flip UVs
 #[derive(Default)]
-pub struct FlipUVs;
+pub struct FlipUVs {
+    first_flip: UvFlipVariant,
+    second_flip: Option<UvFlipVariant>,
+}
+impl FlipUVs {
+    fn process_material(&self, material: &mut AiMaterial) -> Result<(), FlipUVsError> {
+        let mut flips = vec![self.first_flip];
+        if let Some(second_flip) = self.second_flip {
+            flips.push(second_flip)
+        }
+        if let Some(prop) = material.get_property_mut(matkey::_AI_MATKEY_UVTRANSFORM_BASE, None, 0)
+        {
+            let transform = bytemuck::from_bytes_mut::<AiUvTransform>(&mut prop.data);
+            for flip in flips.iter() {
+                match flip {
+                    UvFlipVariant::X => {
+                        transform.translation.x *= -1.0;
+                        transform.rotation *= -1.0;
+                    }
+                    UvFlipVariant::Y => {
+                        transform.translation.y *= -1.0;
+                        transform.rotation *= -1.0;
+                    }
+                    _ => {
+                        transform.rotation *= -1.0;
+                    }
+                }
+            }
+        }
+
+        Ok(())
+    }
+
+    fn process_mesh(&self, mesh: &mut AiMesh) -> Result<(), FlipUVsError> {
+        let mut flips = vec![self.first_flip];
+        if let Some(second_flip) = self.second_flip {
+            flips.push(second_flip)
+        }
+
+        let uv_channel_names = &mesh.texture_coordinate_names;
+
+        let occupied_uv_channels = mesh
+            .texture_coords
+            .iter_mut()
+            .enumerate()
+            .filter_map(|(index, channel_o)| {
+                if channel_o.is_some() {
+                    return Some((index, channel_o.as_mut().unwrap()));
```

Comment:
Minor: use `if let Some(obj) = channel_o.as_mut()` or some equivalent. I prefer this syntax as it frees up unwraps to be in areas wherein there needs to be review that the unwrap conditions are met to avoid panic.